src/fstack.rs

```diff
@@ -51,6 +51,11 @@ impl Node {
     pub fn is_quiet(&self) -> &bool {
         &self.quiet
     }
+
+    /// Whether the node is a `REPT`.
+    pub fn is_rept(&self) -> bool {
+        matches!(self.type_data(), NodeType::Rept(..))
+    }
```

Comment:
Probably a good idea to add similar functions for the other variants?

Reply:
go ahead if you want to